tensorflow_quantum/datasets/spin_system.py

```diff
@@ -0,0 +1,324 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Quantum datasets for quantum many-body spin systems"""
+
+from collections import namedtuple
+import os
+
+import numpy as np
+import sympy
+import cirq
+from tensorflow.keras.utils import get_file
+
+SpinSystemInfo = namedtuple(
+    'SpinSystemInfo',
+    [
+        'g',  # Numpy `float` order parameter.
+        'gs',  # Numpy `complex` array ground state wave function from exact
+        # diagonalization.
+        'gs_energy',  # Numpy `float` ground state energy from exact
+        # diagonalization.
+        'res_energy',  # Python `float` residual between the  circuit energy and
+        # the exact energy from exact diagonalization.
+        'fidelity',  # Python `float` overlap between the circuit state and the
+        # exact ground state from exact diagonalization.
+        'params'  # Dict with Python `str` keys and Numpy`float` values.
+        # Contains $M \times P parameters. Here $M$ is the number of
+        # parameters per circuit layer and $P$ the circuit depth.
+    ])
+
+
+def unique_name():
+    """Generator to generate an infinite number of unique names.
+
+    Yields:
+        Python `str` of the form 'theta_<integer>'
+
+    """
+    num = 0
+    while True:
+        yield 'theta_' + str(num)
+        num += 1
+
+
+def spin_system_data_set(nspins, system_name, data_dir):
+    """Download data and load the data and convert to useful data structres
+
+    Args:
+        nspins: Python `int` number of spins in the system.
+        system_name: Python `str` name of the system.
+        data_dir: Python `str` location where to store the data on disk.
+            Default is `~/tfq-datasets`
+
+    Returns:
+        A Python `lst` of Python `float` order parameters for the system
+        A Python `lst` of `namedtuple` instances containing the following
+            fields:
+            - `g`: Numpy `float` order parameter.
+            - `gs`: Numpy `complex` array ground state wave function from
+                exact diagonalization.
+            - `gs_energy`: Numpy `float` ground state energy from exact
+                diagonalization.
+            - `res_energy`: Python `float` residual between the circuit energy
+                and the exact energy from exact diagonalization.
+            - `fidelity`: Python `float` overlap between the circuit state
+                and the exact ground state from exact diagonalization.
+            - `params`: Dict with Python `str` keys and Numpy`float` values.
+                Contains $M \times P parameters. Here $M$ is the number of
+                parameters per circuit layer and $P$ the circuit depth.
```

Comment:
It would be preferrable if you didn't indent these.

Keep it at the same level as `A Python lst of Python...`

Similar for all other Returns. I see this in other places too.